### PR TITLE
[PM-2192] Improve userkey verification on biometric unlock

### PR DESCRIPTION
**Pull request by quexten**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-2192
https://github.com/bitwarden/clients/issues/5407

## 📔 Objective

Improve userkey validation for biometric unlock by using the cryptoservice's userkey validation functionality & add separate error message for debugging #5407 

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/browser/src/background/nativeMessaging.background.ts [hunk 1]
```diff
@@ -350,7 +362,26 @@ export class NativeMessagingBackground {
...
-              await this.cryptoService.setUserKey(userKey);
+              const activeUserId = await firstValueFrom(
+                this.accountService.activeAccount$.pipe(map((a) => a?.id)),
+              );
+              const isUserKeyValid = await this.cryptoService.validateUserKey(
+                userKey,
+                activeUserId,
+              );
+              if (isUserKeyValid) {
+                await this.cryptoService.setUserKey(userKey);
```
Comment (justindbaur) [inline apps/browser/src/background/nativeMessaging.background.ts:373]: You can now pass `activeUserId` into this method. Ideally we will be making that parameter non-optional soon. Which @trmartin4 is actually doing in #9675.

File: apps/browser/src/background/nativeMessaging.background.ts [hunk 2]
```diff
@@ -350,7 +362,26 @@ export class NativeMessagingBackground {
...
+              } else {
+                this.logService.error("Unable to verify biometric unlocked userkey");
+                await this.cryptoService.clearKeys();
```
Comment (justindbaur) [inline apps/browser/src/background/nativeMessaging.background.ts:376]: Here as well.

**Review by justindbaur — CHANGES_REQUESTED** [review]
Couple minor changes.

File: apps/browser/src/background/nativeMessaging.background.ts (hunk 2 above)
Comment (quexten) [reply apps/browser/src/background/nativeMessaging.background.ts]: Fixed both, thank you!

**Review by justindbaur — APPROVED** [review]

**Comment by quexten** [comment]
Resolved a merge conflict.

**Review by justindbaur — APPROVED** [review]

**Pull request merged**